internal/databases/mongo/delete_handler.go

```diff
@@ -105,7 +105,12 @@ func HandleBackupsPurge(backupTimes []internal.BackupTime,
 		return nil, nil, err
 	}
 
-	purge, retain, err = archive.SplitPurgingBackups(backups, opts.retainCount, opts.retainAfter)
+	timedBackups := archive.MongoModelToTimedBackup(backups)
```

Comment:
Is it necessary? You've already implemented internal.TimedBackup for MongoModelBackup.